### ISSUE TITLE
npm install blocked by other registry referenced in package-lock

### ISSUE THREAD

**Issue by TrevorBurnham** [npm/npm#16973]
#### I'm opening this issue because:

  - [x] npm is doing something I don't understand.

#### What's going wrong?

After upgrading to npm5, `npm install` started freezing in a certain project:

<img width="747" alt="screen shot 2017-06-04 at 6 21 11 pm" src="https://cloud.githubusercontent.com/assets/224895/26766048/5c7a9122-4957-11e7-8fb1-8a273355ce57.png">

After investigating for a while, I realized that I could install if I deleted the `package-lock.json`. The problem was that I'd run an earlier install while using my company's npm registry, and some other dependency had its `version` set to a URL in the company registry. Since accessing the company registry requires a VPN, npm's request to it was receiving no response, blocking the install.

I suppose the `version`-setting behavior is WAD, but why does `npm install <package>` trigger a request involving a package that's totally unrelated to `color-save`? Why isn't that request mentioned in the output of `npm install` with the `--verbose` flag set? And why doesn't the request ever (?) time out if the server doesn't respond?

Also, it might be a good idea for `npm install` to emit a warning when a `package-lock.json` causes it to make a request to a registry that doesn't match `npm config get registry`.  I understand why this is the behavior, but it threw me for a loop initially. I'm really not used to `npm install`'s behavior depending on registries that aren't mentioned in my `.npmrc`.

#### How can the CLI team reproduce the problem?

Create a new directory with this `package-lock.json`:

```json
{
  "name": "my-project",
  "version": "1.0.0",
  "lockfileVersion": 1,
  "dependencies": {
    "is-arrayish": {
      "version": "https://fhgwg.com/is-arrayish/-/is-arrayish-0.3.1.tgz",
      "integrity": "sha1-wt/DhquqDD4zxI2z/ocFnmkGXv0="
    }
  }
}
```

Then run:

```sh
> npm install color-name
```

The install will block on the request to fhgwg.com.

### supporting information:

 - `npm -v` prints: 5.0.2 (also observed in 5.0.0)
 - `node -v` prints: v8.0.0
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: macOS

**Comment by zkat** [comment]
Closing as a duplicate of https://github.com/npm/npm/issues/16728 because of the URL thing. npm5 is otherwise able to adapt to different registries.
